### ISSUE TITLE
2.0: Missing Document's expirationDate implementation

### ISSUE THREAD

**Issue by pasin** [couchbase/couchbase-lite-ios#1870]
Spec: https://github.com/couchbaselabs/couchbase-lite-apiv2/blob/master/spec/spec/Document.swift#L106

**Comment by pasin** [comment]
LiteCore:

```
/** Sets an expiration date on a document.  After this time the
        document will be purged from the database.
        @param db The database to set the expiration date in
        @param docId The ID of the document to set the expiration date for
        @param timestamp The UNIX timestamp of the expiration date (must
                    be in the future, i.e. after the current value of time()).  A value
                    of 0 indicates that the expiration should be cancelled.
        @param outError Information about any error that occurred
        @return true on sucess, false on failure */
    bool c4doc_setExpiration(C4Database *db C4NONNULL,
                             C4String docId,
                             uint64_t timestamp,
                             C4Error *outError) C4API;

    /** Returns the expiration time of a document, if one has been set, else 0. */
    uint64_t c4doc_getExpiration(C4Database *db C4NONNULL, C4String docId) C4API;
```

Does this mean that the document need to be saved first before being able to set the expiration date?

**Comment by snej** [comment]
Currently the expiration times are saved in a separate table from the documents, so I don't think it matters whether you save the doc or set the expiration time first.

**Comment by pasin** [comment]
With the fact that the newly created documents don't have the database assigned until saving time, for that case we need to defer setting the expiration time until when the document is saved.

**Comment by pasin** [comment]
This task is pending for the TTL behavior/feature review.

**Comment by pasin** [comment]
Moved to https://github.com/couchbaselabs/couchbase-lite-apiv2/issues/77 instead.

**Comment by rajagp** [comment]
@pasin, @djpongh   :   Will this potentially make it into 2.1 ?

From the forums : 
https://forums.couchbase.com/t/couchbase-lite-2-0-document-time-to-live/16775

**Comment by djpongh** [comment]
@rajagp unfortunately, it did not.

**Comment by timritzer** [comment]
@pasin 
Any update on this? With this issue closed, and the links having 404s it's hard to tell.

We use this feature heavily to clean up and avoid long client delays in our Enterprise Edition, and just updated to 2.X. The lack of this feature is causing long client delays due to runaway data. The server is TTLing the old data, but clients don't honor it, and no longer have their own corresponding TTL functionality.

Some kind of a timeline or issue to track would be much appreciated.

**Comment by pasin** [comment]
We will have the TTL feature in the coming up release (2.5).

**Comment by timritzer** [comment]
Thanks!

**Comment by snej** [comment]
FYI this was implemented in December — see #2249.
